### ISSUE TITLE
Test for 6 pages took 5GB space in graphite with retention settings 10s:7d, 1d:90d

### ISSUE THREAD

**Issue by vladsoo** [sitespeedio/sitespeed.io#3902]
### Your question

I ran a scripted sitespeed test with 6 pages and used two different retention settings for the Graphite database:

1. 10s:7d, 1d:90d
2. 10s:90d, 30m:360d

After the test, I noticed that the sizes of the Whisper databases for these two runs were significantly different, but both were extremly large. 
The first run resulted in a database size of approximately 5GB, while the second run had a much larger size of around 65GB.

I'm unsure whether these database sizes are considered normal or if there might be an issue. Because even 5GB for such a short period seems too big.
I found this topic a bit relevant to what I have - https://github.com/sitespeedio/sitespeed.io/issues/1416.
But the size discussed in that topic cannot be compared to what I get.
Could you please provide some guidance on what is considered an acceptable size for a Whisper database in relation to the metrics collected and the retention settings used?

Here is the sitespeed config that I used for this tests:
```
{
  "browsertime": {
    "browser": "chrome",
    "viewPort": "1366x768",
    "iterations": 1,
    "visualElements": true,
    "visualMetric": true,
    "spa": true,
    "connectivity":{
    	"profile": "custom",
    	"custom":{
    		"upstreamKbps": 30000,
        	"downstreamKbps": 30000
    	}
    }
  },
  "chrome": {
      "collectLongTasks": true
  },
  
  "graphite": {
  	"host": "172.17.0.1",
  	"namespace": "sitespeed_io",
  	"addSlugToKey": true,
  	"postMetric": "10s"
  },
  "screenshot": {
    "type": "jpg"
  },
  "gzipHAR": true,
  "multi": true,
  "filmstrip": "showAll"
}
```

**Comment by soulgalore** [comment]
Hi @vladsoo yep that sounds too big. I'll help you tomorrow evening when I'm back with more documentation on how Graphite works and how you can set it up.

Best
Peter

**Comment by soulgalore** [comment]
Hi again @vladsoo I think https://www.sitespeed.io/documentation/sitespeed.io/graphite/#configure-graphite is a good start and then https://github.com/statsd/statsd/blob/master/docs/graphite.md for how to setup Graphite.

Graphite allocate the size depending on how you setup the retention, if you setup a rule that match:
`10s:7d, 1d:90d` 
it means in a sitespeed.io case that you gonna test the same URL every ten second and keep that data for  7d. By default we push many metrics and prepare for them every ten seconds will allocate a lot of space. 

When I run my test I usually test every hour (or something like that) depending on how often the site is deployed to production. Then I setup the retention to be something like:
`1h:40d`

Check the documentation and you can see that there's also a pattern match, that means that we can have different retentions depending on the data that is sent to Graphite. In some of the setups I've done, we keep the data for individual runs (default is normally only passing on median/min/max etc). 

Let me update the documentation the coming days so its easier to understand how to setup it correctly.

**Comment by vladsoo** [comment]
Hello @soulgalore!
1. I've read both of those topics before and I believe I understand how retention settings can impact the size, but I'm still a bit puzzled. As I mentioned before, in one of my test runs, I have a 5GB allocated space with the retention set to 10s:7d, 1d:90d. 
Is this size considered normal, or could there be other factors affecting the size besides the retention settings?

2. 
> Check the documentation and you can see that there's also a pattern match, which means that we can have different retentions depending on the data that is sent to Graphite. In some of the setups I've done, we keep the data for individual runs (default is normally only passing on median/min/max, etc.).

Is it about the data types that sitespeed sends to Graphite or am I wrong?
Because next what I thought to reduce the data size is to manipulate with metrics from sitespeed and keep only the most necessary.

**Comment by vladsoo** [comment]
Thanks for pointing out whether it is necessary to use exactly `10s:7d`.
Because defiantly I don't need it that often.
`30m:7d,1d:180d` will be enough, or maybe later I configure it to be even less frequent.
And with this retention policy for the same test with 6 pages, it took only 50 MB.

p.s. I don't remember where I found the information, but I've read something that for sitespeedConfig should be specified option `graphite: {"postMetric": "10s"}`. So I decided if it posts metrics every 10 seconds, Graphite should also be able to gather them every 10s. 
 But it does not correlate.
 
 So thank you a lot!
